bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/adapters/read/ProtoRowAdapter.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.bigtable.hbase.adapters.read;
+
+import com.google.api.core.InternalApi;
+import com.google.bigtable.v2.Cell;
+import com.google.bigtable.v2.Column;
+import com.google.bigtable.v2.Family;
+import com.google.bigtable.v2.Row;
+import com.google.cloud.bigtable.hbase.adapters.ResponseAdapter;
+import com.google.cloud.bigtable.hbase.util.ByteStringer;
+import com.google.cloud.bigtable.hbase.util.TimestampConverter;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.hadoop.hbase.client.Result;
+
+/**
+ * Adapt between a {@link Row} and an hbase client {@link Result}.
+ *
+ * <p>For internal use only - public for technical reasons.
+ */
+@InternalApi("For internal usage only")
+public class ProtoRowAdapter implements ResponseAdapter<Row, Result> {
+
+  @Override
+  public Result adaptResponse(Row row) {
+    if (row == null) {
+      return Result.EMPTY_RESULT;
+    }
+
+    List<org.apache.hadoop.hbase.Cell> hbaseCells = new ArrayList<>();
+    byte[] rowKeyBytes = ByteStringer.extract(row.getKey());
+
+    for (Family family : row.getFamiliesList()) {
+      byte[] familyBytes = ByteStringer.extract(family.getNameBytes());
+
+      for (Column column : family.getColumnsList()) {
+        byte[] qualifierBytes = ByteStringer.extract(column.getQualifier());
+
+        for (Cell cell : column.getCellsList()) {
+          // Cells with labels are treated as internal cells, not sending them to user.
```

Comment:
I thought we agreed that labels need to be able to propagate out of the wrapper and the labels will get filtered out by the whilescanadapter? ie. the work you started here:
https://github.com/googleapis/java-bigtable-hbase/pull/2397

Reply:
Yes, we are going to apply whileMatchFilter for rows returned by readRows() operation(i.e. `ResultScanner<Result>`).

But `readModifyWriteRowAsync()` returns the response directly without any filtering logic, Should we apply the filter on this operation as well. something like
```java
@Override
  public ApiFuture<Result> readModifyWriteRowAsync(ReadModifyWriteRow readModifyWriteRow) {
    ListenableFuture<ReadModifyWriteRowResponse> response =
        delegate.readModifyWriteRowAsync(readModifyWriteRow.toProto(requestContext));
    return ApiFutureUtil.transformAndAdapt(
        response,
        new Function<ReadModifyWriteRowResponse, Result>() {
          @Override
          public Result apply(ReadModifyWriteRowResponse response) {
            return Adapters.BIGTABLE_WHILE_MATCH_FILTER_ADAPTER.adapt(
                protoRowToResultUtil(response.getRow()));
          }
        });
  }
```

Reply:
I dont whilematchfilter & labels make sense for ReadModifyWrite